Add convenience methods for making HTTPTransitions

kylef opened the pull request:
@zdne this greatly simplifies any code examples using the representor as they don't need to worry about this logic. Afterwards an example looks something like as follows:

Given a HTTPTransition:

``` swift
let transition = HTTPTransition(uri: "/notes/{id}", attributes: [:], parameters: [:])
```

Making a request, expanding the URI and encoding any attributes:

``` swift
let baseURL = NSURL(string: "https://api.apiary.io/")
request(baseURL, transition, parameters:["id": "4"], attributes:["key":"value"]).responseJSON { (_, _, JSON, _) in
    println(JSON)
}
```

It makes use of [Alamofire](https://github.com/Alamofire/Alamofire), a lightweight HTTP network requesting.

kylef committed:
Add convenience methods for making HTTPTransitions

kylef commented:
@zdne The test failures are due to some target configuration in Alamofire itself which I'll fix up, but for now please give me a :+1: or :-1: on this.

zdne commented:
:+1: the interface looks good, now my question is whether in 

```
parameters:["id": "4"], attributes:["key":"value"]
```

We can't somehow decouple the knowledge of the parameter name (e.g. not hardcode "id" or "key"). But this is probably out of the scope of this PR.

zdne commented:
@kylef also what is exactly `[:]` in 

```
 attributes: [:]
```

?

kylef commented:
@zdne Certainly, the transition object already has the knowledge of the available keys. `transition.properties` and `transition.attributes` can be used to introspect available keys and dynamically build a UI for them. That was out of the scope for this simple code example in the PR.

kylef commented:
Basically, in Swift both arrays and dictionaries use the square brackets `[]`. The colon inside is shorthand for an empty dictionary `[:]`.

zdne commented:
> Basically, in Swift both arrays and dictionaries use the square brackets []. The colon inside is shorthand for an empty dictionary [:].

Can't this be omitted? It feels a bit tedious (but maybe this was just to demonstrate the use-case).

kylef commented:
@zdne Sure, it would require unrelated changes (making these values optionals with defaults). Outside the scope of this PR as it would change the representer. I'll make another PR.

kylef commented:
@zdne See https://github.com/the-hypermedia-project/representor-swift/pull/10

kylef commented:
Okay, rebased this now that #10 is merged so it can make use of the new optional values in initialisers.

I've also changed to use Alamofire from my own fork to make use of the changes from this [pending pull request](https://github.com/Alamofire/Alamofire/pull/326).

@zdne This should be ready for review/merge now.

zdne commented on Representor.podspec:
```diff
@@ -19,7 +19,7 @@ Pod::Spec.new do |spec|
       core_spec.dependency 'Representor/Core'
       core_spec.dependency 'Representor/Adapter/HAL'
-      core_spec.source_files = 'Representor/HTTP/*.swift', 'Representor/HTTP/Adapters/*.swift'
+      core_spec.source_files = 'Representor/HTTP/HTTP{Transition*,Deserialization}.swift', 'Representor/HTTP/Adapters/*.swift'
```
This is funky :)

kylef replied:
Yeah, gonna restructure this in later pull request. Should be able to break it up more.

zdne commented on Representor.xcodeproj/project.pbxproj:
```diff
@@ -12,6 +12,9 @@
 		276A2C051A7BA4EE004BCC6F /* BlueprintTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 276A2C031A7BA4EE004BCC6F /* BlueprintTests.swift */; };
 		276A2C061A7BA4EE004BCC6F /* BlueprintTransitionTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 276A2C041A7BA4EE004BCC6F /* BlueprintTransitionTests.swift */; };
 		276A2C081A7BA500004BCC6F /* blueprint.json in Resources */ = {isa = PBXBuildFile; fileRef = 276A2C071A7BA500004BCC6F /* blueprint.json */; };
+		276A2C291A7BC0CB004BCC6F /* URITemplate.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 276A2C1E1A7BC07C004BCC6F /* URITemplate.framework */; };
```
I wonder if there is a way to make GitHub ignoring these files (not .gitignore of course)

zdne commented on Representor/HTTP/HTTPRequest.swift:
```diff
@@ -0,0 +1,41 @@
+extension HTTPTransition {
+  var suggestedParameterEncoding:ParameterEncoding {
+    let encodingMapping = [
+      "application/json": ParameterEncoding.JSON,
```
I just wonder, would it make a sense for this to work with JSON based media types in a future `xxx+json`?

kylef replied:
I think it would be better to allow user's to register their own mime types and what it corresponds too.

Although the RFC states to use `application/json` there are still many servers which use the incorrect type such as `text/javascript` and it would be best to leave this up to the user.

zdne commented on Representor/HTTP/HTTPRequest.swift:
```diff
@@ -0,0 +1,41 @@
+}
+
+extension Alamofire.Manager {
+  public func request(baseURL: NSURL?, transition: HTTPTransition, parameters: [String : AnyObject]? = nil, attributes: [String : AnyObject]? = nil, encoding:ParameterEncoding? = nil) -> Alamofire.Request {
```
Nit: This is a looooong line
